### PR TITLE
Remove unneeded `ActionGenerator` class.

### PR DESCRIPTION
**Pull request by clokep**
As far as I can tell this class isn't very useful, it just passes through to `BulkPushRuleEvaluator`. It seems much clearer to remove it and call that directly.

### CODE REVIEW

File: synapse/push/action_generator.py [hunk 1]
```diff
@@ -1,44 +0,0 @@
...
-class ActionGenerator:
-    def __init__(self, hs: "HomeServer"):
-        self.clock = hs.get_clock()
-        self.bulk_evaluator = BulkPushRuleEvaluator(hs)
-        # really we want to get all user ids and all profile tags too,
-        # since we want the actions for each profile tag for every user and
-        # also actions for a client with no profile tag for each user.
-        # Currently the event stream doesn't support profile tags on an
-        # event stream, so we just run the rules for a client with no profile
-        # tag (ie. we just need all the users).
```
Comment (clokep) [inline synapse/push/action_generator.py:38]: It is possible that this comment is useful, but I can't really figure out what it is trying to say?

Comment (erikjohnston) [reply synapse/push/action_generator.py:38]: I have literally no idea wtf this comment is smoking

**Review by erikjohnston — APPROVED** [review]

**Comment by erikjohnston** [comment]
BURN IT ALL DOWN :fire: :fire: :fire: :fire: :fire: :fire: :fire: :fire: :fire: :fire:

File: synapse/push/bulk_push_rule_evaluator.py [hunk 2]
```diff
@@ -185,13 +187,18 @@ async def _get_power_levels_and_sender_level(
...
     async def action_for_event_by_user(
         self, event: EventBase, context: EventContext
     ) -> None:
         """Given an event and context, evaluate the push rules, check if the message
         should increment the unread count, and insert the results into the
         event_push_actions_staging table.
         """
+        if event.internal_metadata.is_outlier():
+            # This can happen due to out of band memberships
+            return
```
Comment (clokep) [inline synapse/push/bulk_push_rule_evaluator.py:200]: @erikjohnston are we OK with this moving into the measured function? It was added in #12689.

Comment (erikjohnston) [reply synapse/push/bulk_push_rule_evaluator.py:200]: Yeah, that's fine!

**Pull request merged**